pkg/bpf/bpf_linux.go

```diff
@@ -58,7 +57,7 @@ type bpfAttrMapOpElem struct {
 // UpdateElementFromPointers updates the map in fd with the given value in the given key.
 func UpdateElementFromPointers(fd int, mapName string, structPtr unsafe.Pointer, sizeOfStruct uintptr) error {
 	var duration *spanstat.SpanStat
-	if option.Config.MetricsConfig.BPFSyscallDurationEnabled {
+	if metrics.BPFSyscallDuration.IsEnabled() {
```

Comment:
We technically don't need the check, if not enabled functions like `Inc`, `Add`, `Observe` are no-op. This was the case before as well so I assume this is an optimization to avoid the cycles used to collect the measurement if we are not going to use it anyway.  No clue if that is warranted or not.